src/data/set/lattice.lean

```diff
@@ -199,52 +221,104 @@ ext $ λ i, mem_Union.symm
 theorem set_of_forall (p : ι → β → Prop) : {x | ∀ i, p i x} = ⋂ i, {x | p i x} :=
 ext $ λ i, mem_Inter.symm
 
-theorem Union_subset {s : ι → set β} {t : set β} (h : ∀ i, s i ⊆ t) : (⋃ i, s i) ⊆ t :=
--- TODO: should be simpler when sets' order is based on lattices
-@supr_le (set β) _ _ _ _ h
+lemma Union_subset {s : ι → set α} {t : set α} (h : ∀ i, s i ⊆ t) : (⋃ i, s i) ⊆ t :=
+@supr_le (set α) _ _ _ _ h
 
-@[simp] theorem Union_subset_iff {s : ι → set β} {t : set β} : (⋃ i, s i) ⊆ t ↔ (∀ i, s i ⊆ t) :=
-⟨λ h i, subset.trans (le_supr s _) h, Union_subset⟩
-
-theorem mem_Inter_of_mem {x : β} {s : ι → set β} : (∀ i, x ∈ s i) → (x ∈ ⋂ i, s i) :=
-mem_Inter.2
+lemma Union₂_subset {s : Π i, κ i → set α} {t : set α} (h : ∀ i j, s i j ⊆ t) :
+  (⋃ i j, s i j) ⊆ t :=
+Union_subset $ λ x, Union_subset (h x)
 
 theorem subset_Inter {t : set β} {s : ι → set β} (h : ∀ i, t ⊆ s i) : t ⊆ ⋂ i, s i :=
 @le_infi (set β) _ _ _ _ h
 
-@[simp] theorem subset_Inter_iff {t : set β} {s : ι → set β} : t ⊆ (⋂ i, s i) ↔ ∀ i, t ⊆ s i :=
-@le_infi_iff (set β) _ _ _ _
+lemma subset_Inter₂ {s : set α} {t : Π i, κ i → set α} (h : ∀ i j, s ⊆ t i j) : s ⊆ ⋂ i j, t i j :=
+subset_Inter $ λ x, subset_Inter $ h x
+
+@[simp] lemma Union_subset_iff {s : ι → set α} {t : set α} : (⋃ i, s i) ⊆ t ↔ ∀ i, s i ⊆ t :=
+⟨λ h i, subset.trans (le_supr s _) h, Union_subset⟩
+
+lemma Union₂_subset_iff {s : Π i, κ i → set α} {t : set α} :
+  (⋃ i j, s i j) ⊆ t ↔ ∀ i j, s i j ⊆ t :=
+by simp_rw Union_subset_iff
+
+@[simp] lemma subset_Inter_iff {s : set α} {t : ι → set α} : s ⊆ (⋂ i, t i) ↔ ∀ i, s ⊆ t i :=
+@le_infi_iff (set α) _ _ _ _
+
+@[simp] lemma subset_Inter₂_iff {s : set α} {t : Π i, κ i → set α} :
+  s ⊆ (⋂ i j, t i j) ↔ ∀ i j, s ⊆ t i j :=
+by simp_rw subset_Inter_iff
 
 theorem subset_Union : ∀ (s : ι → set β) (i : ι), s i ⊆ (⋃ i, s i) := le_supr
 
+lemma subset_Union₂ {s : Π i, κ i → set α} (i : ι) (j : κ i) : s i j ⊆ ⋃ i j, s i j :=
+show s i j ≤ ⨆ i j, s i j,
+  from le_supr_of_le i $ le_supr _ j
```

Comment:
I assume adding an `le_supr₂` lemma is out of scope for this PR?

Reply:
Yup, by the time I fix everything, this one might already be 400lines. I will happily add the missing `supr`/`infi` lemma in a second PR, or in this one if you really want me to.

Reply:
I'd say later because this is a `chore` and not a `feat` PR :wink: